examples/image_decoders/jpeg_viewer.js

```diff
@@ -27,56 +27,58 @@ const jpegCtx = jpegCanvas.getContext("2d");
 
 // Load the image data, and convert it to a Uint8Array.
 //
-let nonBinaryRequest = false;
-const request = new XMLHttpRequest();
-request.open("GET", JPEG_IMAGE, false);
-try {
-  request.responseType = "arraybuffer";
-  nonBinaryRequest = request.responseType !== "arraybuffer";
-} catch (e) {
-  nonBinaryRequest = true;
-}
-if (nonBinaryRequest && request.overrideMimeType) {
-  request.overrideMimeType("text/plain; charset=x-user-defined");
-}
-request.send(null);
+(async function () {
+  let nonBinaryRequest = false;
+  const request = await fetch(JPEG_IMAGE);
+  try {
+    const binaryRequestType = "arraybuffer";
+    nonBinaryRequest = !binaryRequestType;
+  } catch (e) {
+    nonBinaryRequest = true;
+  }
```

Comment:
Exactly! It was confusing me to me too, because on line `30` of the original code, `nonBinaryRequest` is set to `false` and on line `34` `request.responseType = "arraybuffer";`.
Since this file would only load images, I see what you are saying.